### ISSUE TITLE
Support basic assemblies in point-and-click UI

### ISSUE THREAD

**Issue by franknoirot** [KittyCAD/modeling-app#4735]
Allow a basic geometry import and transform workflow without having to hand-write KCL code.

## Prerequisites
1. #4683 
2. #4373 

## Requirements overview
- [x] [1. users can import geometry via point-and-click and see it rendered in the scene](https://github.com/KittyCAD/modeling-app/issues/6120)
- [ ] 2. users can select this imported geometry with a click
- [ ] 3. users can delete this imported geometry without deleting code directly
- [x] 4. [users can transform this imported geometry's position, scale and rotation](https://github.com/KittyCAD/modeling-app/issues/6020)

### 1. users can import geometry via point-and-click and see it rendered in the scene
Notes:
1. Should import be a toolbar button, or like a sidebar button?
2. I think we'll need to augment the `options` argument config to be able to take `async` callbacks if we want to list the current project files as options
3. Is there a way to detect what files in the project actually have exports so that the user's list can be filtered to only usable files?

A proposed user flow for import:
1. Invoke the import command
	1. via project menu item, system menu item, or command palette
2. Allow user to choose what to import
	1. Could use options input with list of all files in the project except the current one
	2. Could open file picker constrained to project folder (is this possible?), but not yet supported as command palette input type
3. Apply code mod to add import statement to top of file
4. Apply code mod to add import invocation to bottom of file
	1. We don't currently have a pattern to apply a code mod in the middle of a command bar flow, so maybe this should be all the import command does
5. (optionally) Allow user to immediately transform the imported part
	1. I'm going to outline this user flow in the second section because I prefer that approach
### 2. users can select this imported geometry with a click
Notes:
1. In NX, once you import a component you can no longer select anything more granular than the entire component (unless you are applying mates), not anything more granular like bodies, faces, edges, or points.
2. It is unclear to me how the artifactGraph will need updated to support selecting imported geometry. @Irev-Dev or @jtran might need to help me out here.
3. Allowing selection via the feature tree should be "free", since imported geometry will be a function invocation of sorts (@jtran can you help catch any misunderstanding I'm having here?)
### 3. users can delete this imported geometry without deleting code directly
Notes:
1. Similar to #4662, we will have to register a code mod for deleting this kind of selection.
2. Should this delete the import statement as well? I think so if it's the last use of that imported geometry

### 4. users can transform this imported geometry's position, scale and rotation
Notes:
1. #4373 is a prerequisite for this being implemented
2. The first pass implementation should not require additional in-scene 3D gizmos or geometry, and try to use simpler command palette inputs
3. A future enhancement issue should be created upon completion of this one to show a 3D transform gizmo in the scene

A proposed first pass implementation:
1. The transform command is invoked, either via command palette, toolbar button, or keybinding
2. Selection argument has a validator that must me one or more `sketch` or `body`
3. Translation argument
	1. Maybe has a new input type of `translate`, shows 3 KCL inputs?
	2. Maybe unroll `translate.x`, `translate.y`, and `translate.z` into separate arguments, each with KCL type?
4. Rotation argument
	1. Same notes as translate argument
5. Scale argument can be a simple KCL-type argument

## Future work
1. users can reference tags, named values, and functions exported by this imported geometry
2. users are shown a gizmo overlay in addition to command palette argument UI flow

**Comment by nadr0** [comment]
> Should import be a toolbar button, or like a sidebar button?

![Image](https://github.com/user-attachments/assets/64d7dd82-30ef-430a-afe6-578861e1bf5d)
^ this maybe too hidden but this is another location for import. 


Best location is probably the side bar in the lower buttons which are not panel buttons.

**Comment by Irev-Dev** [comment]
### 2.
points 1 and 2, I'm not sure how the artifactGraph will look either, since it still all needs to be executed the whole graph could/should be populated, but there's probably not indication that an import even happened since I believe this code import is all kcl (engine import is for non-code assets right), so the artifactGraph probably has no knowledge that an import even happened, that might be more of an argument for moving it into rust.

In regards to NX not allowing edits on imported geometry, I like that, we can add a "right-click->edit" where we open the imported file for them, but I think being strict about these things make sense as the first pass because it likely makes our life easier (i.e. maybe we get cache benefits if we know the imports will never change, only get shifted around).

### 3.
I think it should delete the import statement, one thing I'm a little unclear on though is using multiple things from one import?

### 4.
One thing I'm worried about (only a little) with general transform utils is that the main aim here is to use it for imported geometry, but I assume it will work on any geometry, and this might mean the client side scene needs tweaking, we probably just need to make sure we have updated plane data before entering sketch mode, otherwise the client side sketch could endup sitting where the original sketch was created, not where it is now.

**Comment by nrc** [comment]
> Is there a way to detect what files in the project actually have exports so that the user's list can be filtered to only usable files?

Note that the top-level geometry of a file can also be used without exports. A file without any exports or top-level geometry is useless, so this list will always be all non-empty files. It might be interesting to list what can be used from a file though.

**Comment by nrc** [comment]
> Allow user to choose what to import 

I think it's important that this UI picks what to import as well as where from in one place. So choosing a file isn't enough, the user choose the item from the file to import.

> Should this delete the import statement as well? I think so if it's the last use of that imported geometry

It should, but doing so is a bit interesting. We don't currently track used/unused variables or imports and we'd want to do this statically not dynamically which we definitely don't do at the moment. For now, a simple text search would give a conservative estimate of whether an import is used (doesn't work for glob imports, but that seems fine initially).

**Comment by pierremtb** [comment]
Unblocked now that #4683 is done. Will need an assignee!
